pkg/cmd/operator/operator.go

```diff
@@ -193,6 +193,23 @@ func Run(healthPort, monitoringPort int32, leaderElection bool, leaderElectionID
 	exitOnError(err, "cannot create Integration label selector")
 	selector := labels.NewSelector().Add(*hasIntegrationLabel)
 
+	selectors := cache.SelectorsByObject{
+		&corev1.Pod{}:        {Label: selector},
+		&appsv1.Deployment{}: {Label: selector},
+		&batchv1.Job{}:       {Label: selector},
+		&servingv1.Service{}: {Label: selector},
+	}
+
+	if ok, err := kubernetes.IsAPIResourceInstalled(c, batchv1.SchemeGroupVersion.String(), reflect.TypeOf(batchv1.CronJob{}).Name()); ok && err == nil {
+		selectors = cache.SelectorsByObject{
+			&corev1.Pod{}:        {Label: selector},
+			&appsv1.Deployment{}: {Label: selector},
+			&batchv1.CronJob{}:   {Label: selector},
+			&batchv1.Job{}:       {Label: selector},
+			&servingv1.Service{}: {Label: selector},
+		}
```

Comment:
Maybe:
```go
selectors[&batchv1beta1.CronJob{}] = struct {
	Label labels.Selector
	Field fields.Selector
}{
	Label: selector,
}
```
instead of duplicating the whole initialisation.

Reply:
yes much better! Nice trick! Was struggling to initialize the cache internal Selector before